src/interfaces/records-write.ts

```diff
@@ -951,4 +935,26 @@ export class RecordsWrite implements MessageInterface<RecordsWriteMessage> {
     const attesters = attestationSignatures.map((signature) => Jws.getSignerDid(signature));
     return attesters;
   }
+
+
+  /**
+   * Fetches the initial RecordsWrite of a record.
+   * @returns The initial RecordsWrite if found; `undefined` if the record is not found.
+   */
+  public static async fetchInitialRecordsWrite(
+    messageStore: MessageStore,
+    tenant: string,
+    recordId: string
+  ): Promise<RecordsWrite | undefined> {
+
+    const query = { entryId: recordId };
+    const { messages } = await messageStore.query(tenant, [query]);
```

Comment:
The query filter `entryId: recordId` should always return either 0 or 1 record, unless there is some bad bug. Happy to jump into a call to clarify?

Reply:
Yup, you're right in this case it wouldn't really matter since it's a single filter and that filter points to a single result. Resolving and approving!